# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/FramingChecker.h
# Commit: ddf4012a7652
# Full Hash: ddf4012a7652e36d144fc43bc99335276deeafbe
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-05-22 03:40:58
# Regressor Bug: 965637
# File Overlap Count: 1
# Description:
#   Bug 965637: Move CSP from Principal into Client, part 1: backend changes. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D27654
# ==============================================================================

diff -r 6c2047bc2f3e -r ddf4012a7652 dom/security/FramingChecker.h
--- a/dom/security/FramingChecker.h	Tue May 21 22:51:54 2019 +0000
+++ b/dom/security/FramingChecker.h	Tue May 21 23:14:27 2019 +0000
@@ -12,14 +12,14 @@
 class nsIHttpChannel;
 class nsIDocShellTreeItem;
 class nsIURI;
-class nsIPrincipal;
+class nsIContentSecurityPolicy;
 
 class FramingChecker {
  public:
   // Determine if X-Frame-Options allows content to be framed
   // as a subdocument
   static bool CheckFrameOptions(nsIChannel* aChannel, nsIDocShell* aDocShell,
-                                nsIPrincipal* aPrincipal);
+                                nsIContentSecurityPolicy* aCSP);
 
  protected:
   enum XFOHeader { eDENY, eSAMEORIGIN, eALLOWFROM };